enumeratum-core/src/main/scala/enumeratum/EnumEntry.scala

```diff
@@ -61,4 +61,17 @@ object EnumEntry {
     abstract override def entryName: String = super.entryName.toLowerCase
   }
 
+  /**
+   * Helper implicit for more readable checking.
+   */
+  implicit class ComparableEnum[A <: EnumEntry](enum: A) {
```

Comment:
We should probably extend from `AnyVal` to [avoid runtime overhead](http://docs.scala-lang.org/overviews/core/value-classes.html).